extra/torch_backend/backend.py

```diff
@@ -40,15 +40,54 @@ def topk(self, k, dim=-1, largest=True, sorted=True):
 @torch.library.impl("aten::_index_put_impl_", "privateuseone")
 def _index_put_impl_(self, indices, values, accumulate=False, unsafe=False):
   # TODO: move to tinygrad
-  return aten._index_put_impl_(self.cpu(), [x.cpu() for x in indices], values.cpu(), accumulate, unsafe).tiny()
+  return aten._index_put_impl_(self.cpu(), [x.cpu() if isinstance(x, torch.Tensor) else None for x in indices], values.cpu(), accumulate, unsafe).tiny()
 
 @torch.library.impl("aten::index.Tensor", "privateuseone")
 def index_tensor(x, y):
   return aten.index(x.cpu(), [z.cpu() if isinstance(z, torch.Tensor) else None for z in y]).tiny()
 
+@torch.library.impl("aten::index_put", "privateuseone")
+def index_put(self, indices, values, accumulate=False):
+  return aten.index_put(self.cpu(), [z.cpu() if isinstance(z, torch.Tensor) else None for z in indices], values.cpu(), accumulate).tiny()
+
+@torch.library.impl("aten::cumprod", "privateuseone")
+def cumprod(self, dim, dtype=None):
+  # TODO: implement in tinygrad
+  return aten.cumprod(self.cpu(), dim, dtype=dtype).tiny()
+
 @torch.library.impl("aten::randperm.generator_out", "privateuseone")
 def randperm_generator(n, generator=None, out=None): out.copy_(torch.randperm(n, generator=generator, device="cpu").tiny())
 
+def upsample_1d_backward(grad_out, output_size, input_size, scales=None, f=None):
```

Comment:
I could try to write custom backward for these, but it could get ugly, also decomps don't exist for these so it's kinda annoying. Any suggestions?